lms/djangoapps/grades/new/subsection_grade.py

```diff
@@ -166,6 +168,21 @@ def create(self, subsection, course_structure, course):
             self._compute_and_save_grade(subsection, course_structure, course)
         )
 
+    def update(self, usage_key, course_key):
+        """
+        Updates the SubsectionGrade object for the student and subsection.
+        """
+        from courseware.courses import get_course_by_id  # avoids circular import with courseware.py
+        course = get_course_by_id(course_key, depth=0)
+        # save ourselves the extra queries if the course does not use subsection grades
+        if not course.enable_subsection_grades_saved:
+            return
+
+        course_structure = get_course_blocks(self.student, usage_key)
```

Comment:
Side note: we should consider caching the result of `get_course_blocks` in the request.